model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -401,6 +425,86 @@ async def create_deepspeed_bundle(
                 )
             ).model_bundle_id
 
+    async def create_vllm_bundle(
+        self,
+        user: User,
+        model_name: str,
+        framework_image_tag: str,
+        endpoint_unique_name: str,
+        num_shards: int,
+        checkpoint_path: Optional[str],
+    ):
+        command = []
+
+        max_num_batched_tokens = 2560  # vLLM's default
+        if "llama-2" in model_name:
+            max_num_batched_tokens = 4096  # Need to be bigger than model's context window
+
+        assert checkpoint_path is not None, "Checkpoint path must be provided for vLLM"
```

Comment:
Should this be `Optional` if we're really requiring it?

Reply:
hmm it might be convenient for external users to make it actually optional